packages/devtools_app/lib/src/ui/hover.dart

```diff
@@ -215,29 +221,124 @@ class HoverCard {
   }
 }
 
+/// Ensures that only one [HoverCard] is ever displayed at a time.
+class HoverCardTooltipController {
+  /// The card that is currenty being displayed
+  HoverCard? _hoverCard;
+
+  /// A new token is given to each [HoverCard].
+  ///
+  /// If [_freshnessToken] matches the [HoverCard.freshnessToken] then it is the
+  /// most recently displayed [HoverCard].
+  int _freshnessToken = 0;
+
+  /// Sets [hoverCard] as the most recently displayed [HoverCard].
+  ///
+  /// This methods sets a unique [hoverCard.freshnessToken], as a way to track whether
+  /// [hoverCard] is the most recently displayed [HoverCard].
+  void set({required HoverCard hoverCard}) {
+    _hoverCard?.remove();
+    hoverCard.freshnessToken = _newFreshnessToken();
+    _hoverCard = hoverCard;
+  }
+
+  /// If the mouse is outside of [_hoverCard] then then it will be removed.
+  void maybeRemoveHoverCard() {
+    final wasRemoved = _hoverCard?.maybeRemove();
+    if (wasRemoved == true) {
+      // Only one hovercard is ever shown on screen. So if the
+      // mouse has moved outside, then no other hovercards are meant to pop up
+      // any more. Invalidating the freshness token means all current hover
+      // cards, that are still generating data, will exit quietly once the data
+      // is available.
+      _invalidateFreshnessToken();
+      _hoverCard = null;
+    }
+  }
+
+  /// Remove the [HoverCard] being displayed, if there is one.
+  void removeHoverCard() {
+    _invalidateFreshnessToken();
+    _hoverCard?.remove();
+    _hoverCard = null;
+  }
+
+  /// Create increment the freshness token and return the new value.
+  int _newFreshnessToken() {
+    return ++_freshnessToken;
+  }
+
+  /// Helper for incrementing the freshness token.
+  void _invalidateFreshnessToken() {
+    _newFreshnessToken();
+  }
+
+  /// Check's that a [hoverCard.freshnessToken] is the same as the active
+  /// freshness token.
+  bool isHoverCardStillFresh(HoverCard hoverCard) {
+    return hoverCard.freshnessToken == _freshnessToken;
```

Comment:
instead of using tokens, you can just have a WeakRef<HoverCard> for the currently active HoverCard. I am assuming you aren't storing the active HoverCard due to concerns about leaking memory if it was stored here.